### PR TITLE
deprecation(semver): comparator min and max properties

### PR DESCRIPTION
**Pull request by timreichen**
deprecate `SemVerComparator` `min` and `max` properties.

These properties do not represent anything in a comparator but are pre-calculated values.
They get obstructive, if one creates a comparator as an object and not parsing a string:

before:
```ts
import { MAX } from "deno.land/std/semver/constants.ts"
const comparator = {
  operator: ">=",
  semver: { major: 1, minor: 2, patch: 3, prerelease: [], build: [] },
  min: { major: 1, minor: 2, patch: 3, prerelease: [], build: [] },
  max: MAX,
}
```
after:
```ts
const comparator = {
  operator: ">=",
  semver: { major: 1, minor: 2, patch: 3, prerelease: [], build: [] } 
}
```

### CODE REVIEW

File: semver/types.ts [hunk 1]
```diff
@@ -49,8 +49,14 @@ export type FormatStyle =
...
   operator: Operator;
   semver: SemVer;
-  min: SemVer;
-  max: SemVer;
+  /**
+   * @deprecated (will be removed in 0.215.0) use {@linkcode comparatorMin} instead.
+   */
+  min?: SemVer;
+  /**
+   * @deprecated (will be removed in 0.215.0) use {@linkcode comparatorMax} instead.
```
Comment (iuioiua) [inline semver/types.ts:57]:
```suggestion
   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMin} instead.
   */
  min?: SemVer;
  /**
   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMax} instead.
```

File: semver/types.ts [hunk 2]
```diff
@@ -76,3 +82,6 @@ export interface SemVerRange {
   // The outer array is OR while each inner array is AND
   ranges: SemVerRangeOr;
 }
+
+export type Range = [SemVerComparator, SemVerComparator];
+export type RangeSet = Range[];
```
Comment (iuioiua) [inline semver/types.ts:87]: Is this change related to the main objective of this PR?

File: semver/parse_comparator.ts [hunk 3]
```diff
@@ -20,12 +18,5 @@ export function parseComparator(comparator: string): SemVerComparator {
...
   const semver = m[2] ? parse(m[2]) : ANY;
-  const min = comparatorMin(semver, operator);
-  const max = comparatorMax(semver, operator);
-  return {
-    operator,
-    semver,
-    min,
-    max,
-  };
+  return { operator, semver };
```
Comment (iuioiua) [inline semver/parse_comparator.ts:21]: This is a breaking change.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Can you please revert any changes that are either breaking or unrelated?

File: semver/types.ts (hunk 2 above)
Comment (timreichen) [reply semver/types.ts]: sorry, this slipped in. reverted.

File: semver/parse_comparator.ts (hunk 3 above)
Comment (timreichen) [reply semver/parse_comparator.ts]: I reverted the breaking change.

File: semver/stringify_range.ts
Comment (iuioiua) [inline semver/stringify_range.ts:1]: Is this related to the PR?

Comment (timreichen) [reply semver/stringify_range.ts:1]: Sorry, this change also slipped in. Reverted

File: semver/types.ts [hunk 4]
```diff
@@ -49,8 +49,14 @@ export type FormatStyle =
...
-  min: SemVer;
-  max: SemVer;
+  /**
+   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMin} instead.
+   */
+  min?: SemVer;
+  /**
+   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMax} instead.
+   */
+  max?: SemVer;
```
Comment (iuioiua) [inline semver/types.ts:59]:
```suggestion
  min: SemVer;
  /**
   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMax} instead.
   */
  max: SemVer;
```
Sorry, I just noticed these were made optional. I think that should be reverted.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Sorry, one last suggestion.

**Review by iuioiua — CHANGES_REQUESTED** [review]
These changes seem to include the deprecation of `SemVerComparator`. Can you please ensure the PR only makes a single change?

**Comment by iuioiua** [comment]
Also, I'm not certain of this change. Either way, let's see what others think.

**Comment by timreichen** [comment]
> These changes seem to include the deprecation of `SemVerComparator`. Can you please ensure the PR only makes a single change?

But `SemVerComparator` was deprecated in https://github.com/denoland/deno_std/pull/3957/files#diff-8dc159a3b198ba82f12b082d1fcb076072d5ff8f10648744ae4746446d749022

**Review by kt3k — APPROVED** [review]
LGTM

File: semver/types.ts [hunk 5]
```diff
@@ -49,7 +49,13 @@ export type FormatStyle =
 export interface Comparator {
   operator: Operator;
   semver: SemVer;
+  /**
+   * @deprecated (will be removed in 0.212.0) use {@linkcode comparatorMin} instead.
+   */
   min: SemVer;
+  /**
```
Comment (iuioiua) [inline semver/types.ts:56]: Actually, I don't think we need to include the deprecation in the new type.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by kt3k — APPROVED** [review]

**Review by iuioiua — APPROVED** [review]
